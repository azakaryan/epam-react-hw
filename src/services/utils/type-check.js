import { Observable } from 'rxjs';

export class TypeCheck {
  static isArray(data) {
    return Object.prototype.toString.call(data) === '[object Array]';
  }

  static isObject(data) {
    return Object.prototype.toString.call(data) === '[object Object]';
  }

  static isString(data) {
    return typeof data === 'string';
  }

  static isFunction(data) {
    return data && {}.toString.call(data) === '[object Function]';
  }

  static isBoolean(data) {
    return typeof data === 'boolean';
  }

  static isNumber(data) {
    return typeof data === 'number';
  }

  static isObservable(data) {
    return data instanceof Observable;
  }
}
