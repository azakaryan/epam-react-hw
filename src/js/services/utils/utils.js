import {TypeCheck} from "./type-check";

export default class Utils {

    static removeblankAttributesfromObject(obj, all = false) {
      for (const propName in obj) {
          const val = obj[propName];
          if (val === null || val === undefined || val === '' || all && this.isEmptyArray(val) || this.isEmptyObject(val)) delete obj[propName];
      }
      return obj;
    }

    static isEmptyObject(item) {
      return TypeCheck.isObject(item) && this.isObjectEmpty(item);
    }

    static isEmptyArray(item) {
      return TypeCheck.isArray(item) && item.length === 0;
    }

    static isObjectEmpty(obj) {
      for (const key in obj) {
        if (obj.hasOwnProperty(key)) return false;
      }
      return true;
    }

}