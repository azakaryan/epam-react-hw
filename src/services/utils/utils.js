import { TypeCheck } from './type-check';

export default class Utils {
  static removeBlankAttributesFromObject(obj, all = false) {
    for (const propName in obj) {
      const val = obj[propName];
      if (val === null || val === undefined || val === '' || all && (this.isEmptyArray(val) || this.isEmptyObject(val))) delete obj[propName];
    }
    return obj;
  }

  static isEmptyObject(obj) {
    if (!TypeCheck.isObject(obj)) return false;
    for (const key in obj) {
      if (obj.hasOwnProperty(key)) return false;
    }
    return true;
  }

  static isEmptyArray(item) {
    return TypeCheck.isArray(item) && item.length === 0;
  }
}
