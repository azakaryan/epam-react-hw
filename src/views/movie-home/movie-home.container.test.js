import React from 'react';
import { shallow } from 'enzyme';
import expectEnzymify from 'expect';
import enzymify from 'expect-enzyme';
import MovieHome, { mapDispatchToProps } from './movie-home.container';
import { TypeCheck } from '../../services/utils/type-check';

expectEnzymify.extend(enzymify());
const onLoad = () => {};

describe('MovieHome Container Component', () => {
  it('should render correctly', () => {
    const component = shallow(<MovieHome onLoad={onLoad} />);
    expect(component).toMatchSnapshot();
  });

  it('should not dispatch an action', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).onLoad();
    expect(dispatch.mock.calls[0][0]('', () => (new Map([['movies', []]])))).toEqual(undefined);
  });

  it('should dispatch an action', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).onLoad();
    expect(dispatch.mock.calls[0][0]('', () => (new Map([['movies', [{}, {}]]])))).toEqual(undefined);
  });

  it('should MovieHome has initialAction static method defined', () => {
    expect(TypeCheck.isFunction(MovieHome.initialAction)).toBeTruthy();
  });

  it('call MovieHome has initialAction static method defined', () => {
    expect(TypeCheck.isFunction(MovieHome.initialAction())).toBeTruthy();
  });
});
