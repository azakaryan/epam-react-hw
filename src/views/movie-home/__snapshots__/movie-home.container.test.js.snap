// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieHome Container Component should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
